walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,117 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      // meeting request has: name, duration, and attendees (List of Strings)
+      // event has: time range, attendees, and name
+
+      long duration = request.getDuration();
+
+       // meeting request exceeds one day
+      if (duration > TimeRange.WHOLE_DAY.duration()) {
+          return Arrays.asList();
+      }
+
+      // all attendees are free
+      if (events.isEmpty()) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+
+      // there are no attendees
+      if (request.getAttendees().isEmpty()) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+      
+
+      Collection<String> mandatory = request.getAttendees();
+      // Collection<String> optional = request.getOptionalAttendees();
```

Comment:
Don't have commented out code. Also see:
https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules

The premise is the YAGNI principle:
https://www.martinfowler.com/bliki/Yagni.html